qiskit/result/counts.py

```diff
@@ -180,3 +185,10 @@ def int_outcomes(self):
                 int_key = int(bitstring.replace(" ", ""), 2)
                 out_dict[int_key] = value
             return out_dict
+
+    def __getitem__(self, key):
```

Comment:
See the comment on L21 we shouldn't be doing this here it will lead to weird behavior. If we want to do this (which I personally disagree with) we should change the parent here to be defaultdict.

But that being said the purpose of this is not clear to me. The count class is a dict subclass that contains entries for each measurement outcome from an execution. The set of things outside that is infinite and I think a keyerror is the correct response. This also enables incorrect behavior such as:
`Counts()['My Incorrect Answer']` which will return 0.

Reply:
removed in 9c976307e26d49431c619936cffae76606c1b5c2